src/darwin/Framework/CHIPTests/MTRPerControllerStorageTests.m

```diff
@@ -3692,6 +3692,11 @@ - (void)testMTRDeviceDealloc
         XCTAssertEqual(controller.devices.count, 1);
     }
 
+    // Wait for the report processing to finish, so the MTRDevice object may dealloc
+    [controller syncRunOnWorkQueue:^{
+        ;
+    } error:nil];
+
```

Comment:
In case the comment task doesn't work to prevent merging, opening a conversation here as well.

Please do not resolve this conversation.